src/material/paginator/paginator.ts

```diff
@@ -54,6 +57,25 @@ export class PageEvent {
   length: number;
 }
 
+
+/** Object that can be used to configure the default options for the paginator module. */
+export interface MatPaginatorConfig {
+  /** Number of items to display on a page. By default set to 50. */
+  pageSize?: number;
+
+  /** The set of provided page size options to display to the user. */
+  pageSizeOptions?: number[];
+
+  /** Whether to hide the page size selection UI from the user. */
+  hidePageSize?: boolean;
+
+  /** Whether to show the first/last buttons UI to the user. */
+  showFirstLastButtons?: boolean;
+}
+
+/** Injection token that can be used to provide the default options for the paginator module. */
+export const MAT_PAGINATOR_CONFIG = new InjectionToken<MatPaginatorConfig>('MAT_PAGINATOR_CONFIG');
```

Comment:
What's the breakdown of naming for `X_DEFAULT_OPTIONS` and `X_CONFIG`?

Reply:
I don't think there's any logic behind it. I've named it `X_CONFIG` in a few other places so that's why I used it here as well.

Reply:
I was how many of each we had. I want to make them all consistent at some point and it would be nice to just start using the one we want to use everywhere. We have stuff like `MAT_DIALOG_DEFAULT_OPTIONS` since dialog has a config object, but autocomplete follows the same pattern. Tabs, though, use `_CONFIG`. Any thoughts on which one should be the ultimate pattern?

Reply:
I don't feel strongly about either one. Maybe the `_CONFIG` because it's shorter?

Reply:
Based on the discussion in Slack, let's gowith `_DEFAULT_OPTIONS` since `_CONFIG` conflicts with some other classes e.g. `MatDialogConfig` and `MAT_DIALOG_CONFIG` would be confusing.